core/src/main/java/org/mule/runtime/core/api/scheduler/SchedulerService.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.scheduler;
+
+import org.mule.runtime.api.service.Service;
+import org.mule.runtime.core.api.processor.ReactiveProcessor.ProcessingType;
+
+/**
+ * Provides access to the different schedulers and thread pools that exist in the Mule runtime, allowing an artifact to schedule
+ * tasks on those.
+ * <p>
+ * This interface provides access to different schedulers, each with its own configured tuning options (TODO MULE-10585),
+ * optimized for some specific kind of work. Artifacts that need to dispatch tasks need to carefully determine which scheduler
+ * best fits the nature of the task in order to keep the resources usage and the impact on other artifacts at a minimum.
+ * <p>
+ * The {@link Scheduler}s returned by methods in the implementations must provide access the the thread pools in the Mule Runtime
+ * that have the expected tuning configuration. Each {@link Scheduler} will have its own lifecycle, managed by its user and NOT
+ * this service.
+ *
+ * @since 4.0
+ */
+public interface SchedulerService extends Service {
```

Comment:
i have to do some implementation for the unit tests in core, that will not need lifecycle.